File: backends/ruby-rails/app/domain/staff/condition.rb
# frozen_string_literal: true
#
# スタッフ検索条件を表すドメインオブジェクトモジュール。
#
# @author Satoshi Nagashiba <satoshi.nagashiba@gmail.com>

module Domain
  module Staff
    # スタッフ一覧取得の検索条件です。
    Condition = Struct.new(:keyword, :roles, :offset, :limit, :sort, :sort_type, keyword_init: true)
  end
end

File: backends/rust-axum/src/domain/staff/condition.rs
//! スタッフドメイン 検索条件モジュール。
//!
//! # Author
//! Satoshi Nagashiba <satoshi.nagashiba@gmail.com>

/// スタッフ一覧の検索条件。
#[derive(Clone)]
pub struct Condition {
    pub keyword:   Option<String>,
    pub roles:     Vec<i32>,
    pub offset:    i64,
    pub limit:     i64,
    pub sort:      Option<String>,
    pub sort_type: Option<String>,
}
